src/OrchardCore/OrchardCore.ReCaptcha.Core/ServiceCollectionExtensions.cs

```diff
@@ -13,12 +13,12 @@ public static class ServiceCollectionExtensions
     {
         public static IServiceCollection AddReCaptcha(this IServiceCollection services, Action<ReCaptchaSettings> configure = null)
         {
-            services.AddHttpClient<ReCaptchaClient>()
+            services.AddScoped<ReCaptchaService>()
+                .AddHttpClient<ReCaptchaService>()
```

Comment:
If the typed client is now tied to `ReCaptchaService`, because it is also injected by `ReCaptchaLoginFilter` on each request, an `HttpClient` will be still created, by using the `IHttpClientFactory` but still on each request.

I will add to #14348  the simple fix I did in #14335 because they are both related to #14117, I may change the lifetime of `ReCaptchaService`, at least a scoped service, maybe a singleton if it no longer injects the client, will see.

But not incompatible with what you did here because other changes and cleanups look good.